README.md

```diff
@@ -9,57 +9,75 @@
 This library allows you to split up large binary messages into multiple
 chunks of a certain size.
 
-When converting data to chunks, a 9 byte header is prepended to each
-chunk. This allows you to send the chunks to the receiver in any order.
+It allows you to choose between the following two modes:
+
+* **Reliable/Ordered**: Intended for reliable and ordered transmission
+  of chunks. Chunks may not be reordered and chunks of different
+  messages may not be interleaved.
+* **Unreliable/Unordered**: Intended for transmission of chunks where
+  chunks may be lost or reordered. **Important**: Duplication of chunks
+  is not allowed by this implementation.
```

Comment:
Is an unchunker for an unreliable connection even useful if it doesn't handle duplicates? After all, if a packet is lost the sender doesn't always know whether the receiver received it or not. And the API does not allow querying which chunks are still missing, right?

Wouldn't that be an implementation for Reliable/Unordered?

Reply:
> Is an unchunker for an unreliable connection even useful if it doesn't handle duplicates?

SCTP (and thus data channels) don't allow for duplicated messages in unreliable/unordered mode. Certainly, you wouldn't be able to use this for UDP any more.

> After all, if a packet is lost the sender doesn't always know whether the receiver received it or not.

Pardon me but isn't that the point of an unreliable transport? :grinning: 

> Wouldn't that be an implementation for Reliable/Unordered?

No, you can still omit chunks.

---

Anyhow, I have an idea how this can be fixed easily.

Reply:
Wouldn't be that easy to implement... and since we don't have a use case where chunk duplication would need to be handled, let's just ignore it for now.

Reply:
> Just curious: How would you recover from a situation where you use an unreliable channel and lose some chunks?

Depends on what you mean by *recovering*. If you're in *unreliable* mode, it means messages can be lost during transmission and the application is fine with that. If the application would not be fine with losing messages, then what you want is a reliable transport.

> How does SCTP handle duplicates? Does it know whether a message was lost at the other end?

Sure, SCTP needs to handle duplicates and will discard duplicated [`DATA` chunk](https://en.wikipedia.org/wiki/SCTP_packet_structure#DATA_chunk)s. Since it's reliable by default, it needs to know whether a `DATA` chunk has been lost. It does so by adding a *transmission sequence number* to each `DATA` chunk. (Which is why it's so absurd that we need to do the same on JS layer again as SCTP will most likely do it much more efficiently.)

Reply:
> Thus, chunking increases the unreliability of a channel.

Yes and no. The probability to lose a message pretty much stays the same since SCTP also needs to chunk the data and the probability increases with each `DATA` chunk being transmitted. Unless you use partial reliability where it's of course a different story.